src/DungeonMap.cpp

```diff
@@ -272,20 +272,27 @@ void CDungeonMap::FillArea( const Uint8 type, JRect *rcFill, const int direction
 
     int x, y;
     JIVector vCurPos( vOrigin );
-    for( y = rcLocal.top; y <= rcLocal.bottom; y++ )
+    lightArea( type, rcLocal, rcEdges, vOrigin );
```

Comment:
If you are lighting the whole area, just send edges but run the whole rect instead of going all the math. 

No need for local